gluu-server/opt/gluu/jetty/oxauth/custom/scripts/sic/oob.py

```diff
@@ -114,6 +116,10 @@ def AuthenticateOutOfBand(self, requestParameters):
             addMessage("oob:resend", FacesMessage.SEVERITY_INFO, "sic.newCode")
 
         if codeExpired or requestParameters.containsKey("oob:resend"):
+            telemetry["result"] = "failed"
+            telemetry["reason"] = "code expired" if codeExpired else "new code requested"
+            self.telemetryClient.trackEvent("OOB Authentication", telemetry, {"duration": duration})
```

Comment:
would be good to mention what time unit this duration is, seconds?

Reply:
Are you suggesting a name change (e.g. durationInSeconds) or just some documentation? This is essentially the think time, i.e. the time the user needed to complete that particular task.

I'm not enamored with the variable name "telemetry" which I've used throughout either. It is really the collection of  event properties. Maybe should be "eventProperties" instead?

Reply:
the name change will help when looking at the event log and the metric unit. as for telemetry, yeah it is quite generic but I think good enough to understand it is a temporary object holder for observabiity